packages/core/src/components/side-pane/side-pane.tsx

```diff
@@ -0,0 +1,361 @@
+      }
+    }
+
+    this.expandedChange.emit({
+      position: this.position,
+      expanded: this.expandPane,
+    });
+  }
+
+  @Listen('resize', { target: 'window' })
+  onWindowResize() {
+    const newMode = window.innerWidth <= 767;
+    if (this.mobile != newMode) {
+      this.mobile = newMode;
+    }
+  }
+
+  @Watch('mobile')
+  onMobileChange(): void {
+    this.resetLayoutState();
+    this.setIcons();
+    this.expandPane = false;
+  }
+
+  resetLayoutState() {
+    this.previousWidth = null;
+    this.previousHeight = null;
+    this.showContent = false;
+    this.hostElement.style.removeProperty('height');
+    this.hostElement.style.removeProperty('width');
+  }
+
+  private animateVerticalFadeIn(size: string) {
+    requestAnimationFrame(() => {
+      anime({
+        targets: this.hostElement,
+        duration: ANIMATION_DURATION,
+        width: size,
+        easing: 'linear',
+        begin: () => {
+          if (!this.expandPane) {
+            this.showContent = false;
+          }
+        },
+        complete: () => {
+          if (this.expandPane) {
+            this.showContent = true;
+          } else {
+            this.hostElement.style.removeProperty('width');
+          }
+        },
+      });
+    });
+  }
+
+  private animateHorizontalFadeIn(size: string) {
+    requestAnimationFrame(() => {
+      anime({
+        targets: this.hostElement,
+        duration: ANIMATION_DURATION,
+        height: size,
+        easing: 'linear',
+        begin: () => {
+          if (!this.expandPane) {
+            this.showContent = false;
+          }
+        },
+        complete: () => {
+          if (this.expandPane) {
+            this.showContent = true;
+          } else {
+            this.hostElement.style.removeProperty('height');
+          }
+        },
+      });
+    });
+  }
+
+  hiddePane(): boolean {
+    if (this.inline) {
+      return false;
+      //floating and mobile --> mobile panes at top and bottom
+    } else if (this.floating && this.mobile) {
+      return false;
+      //floating at normal screen --> panes are hidden if they are not expanded
+    } else if (this.floating && this.expandPane && !this.mobile) {
+      return false;
+    }
+
+    return true;
+  }
+
+  hidePaneContent() {
+    if (this.showPreviewContent && this.isLeftRightPane && !this.mobile) {
+      return false;
+    } else if (this.showContent) {
+      return false;
+    }
+    return true;
+  }
+
+  render() {
+    return (
+      <Host
+        hidden={this.hiddePane()}
+        class={{
+          'mobile-overlay': this.expandPane && this.mobile,
+          'top-expand': this.expandPane && this.isMobileTop && this.mobile,
+          'bottom-expand': this.expandPane && !this.isMobileTop && this.mobile,
+          'preview-content':
+            this.isLeftRightPane &&
+            !this.expandPane &&
+            !this.mobile &&
+            this.showPreviewContent,
+        }}
+      >
```

Comment:
aria labels are missing. Adapt to one of the nearest pattern https://www.w3.org/WAI/ARIA/apg/patterns/